syntax = "proto3";

import "ClientInfo.proto";

/*
 * A message sent by Host to Client via wireless connection.
 */
message WirelessHostMessage {
  enum Type {
    PING = 0;  // A reply to a Ping message.
    HANDSHAKE_ASSIGNMENT = 1;  // A reply to a Handshake message.
    RESET_REQUEST = 2;  // A request for clients to reset.
    ARBITRATION_CONFIRMATION = 3;  // A confirmation of successful arbitration.
  }
  Type type = 1;  // The message type.

  oneof content {
    ClientInfo assignment = 2;
  }
}
